src/data/matrix/basic.lean

```diff
@@ -82,15 +84,6 @@ instance [subsingleton α] : subsingleton (matrix m n α) := pi.subsingleton
 instance [nonempty m] [nonempty n] [nontrivial α] : nontrivial (matrix m n α) :=
 function.nontrivial
 
-@[simp] theorem zero_apply [has_zero α] (i j) : (0 : matrix m n α) i j = 0 := rfl
-@[simp] theorem neg_apply [has_neg α] (M : matrix m n α) (i j) : (- M) i j = - M i j := rfl
-@[simp] theorem add_apply [has_add α] (M N : matrix m n α) (i j) :
-  (M + N) i j = M i j + N i j :=
-rfl
-@[simp] theorem sub_apply [has_sub α] (M N : matrix m n α) (i j) :
-  (M - N) i j = M i j - N i j :=
-rfl
-
```

Comment:
What's the rationale for removing these?

Reply:
They're done in `data.matrix.dmatrix` now. If you leave these in the `simp` linter complains.

Reply:
Unfortunately it's difficult to actually define `matrix` in terms of `dmatrix`. I tried, and it ... mostly works, but the exceptions seemed pretty painful.

Reply:
Oh, so `simp` sees through the `matrix` and `dmatrix` definitions?

Reply:
I'm pretty confused about this --- it manages to see through the `matrix` definition to use the `dmatrix` lemmas, but can't see through `dmatrix` to use the `pi` `apply` lemmas. Perhaps its because there are two variables somehow.